### ISSUE TITLE
Fix parent-child relationships for some Qt objects

### ISSUE THREAD

**Issue by elfring** [CSMathematics/DataTex#7]
:eyes: Some Qt objects are created without specifying a corresponding pointer for the parameter “parent”.

Examples:
* BackUp constructor
  https://github.com/CSMathematics/DataTex/blob/99fda6b887907f2e8e6b8f2425b09b0c7df1ccec/datatex/backup.cpp#L63-L65
* DataTables::LoadFields
  https://github.com/CSMathematics/DataTex/blob/a6eff0873df20c42df2ff33eb7f0554f53e625a0/datatex/datatables.cpp#L278-L283
* InfoPage constructor
  https://github.com/CSMathematics/DataTex/blob/a6eff0873df20c42df2ff33eb7f0554f53e625a0/datatex/basefolder.cpp#L75-L93
* LatexTextBrowser constructor
  https://github.com/CSMathematics/DataTex/blob/99fda6b887907f2e8e6b8f2425b09b0c7df1ccec/datatex/highlighter.cpp#L148-L152

:thought_balloon: I got the impression that such data would be required for [safe management of parent-child relationships](https://doc.qt.io/qt-6/objecttrees.html "Object Trees & Ownership in Qt") at more source code places.

**Comment by CSMathematics** [comment]
Indeed, there are many parts of code that need to be carefully considered.Τhanks for the remark .
